### PR TITLE
Student sign out

### PR DESCRIPTION
**Pull request by jenmccarthy**
I would just like a bit of feedback as I am still going to implement a signed_in_time for an attendance record. 

I'm not sure if I've covered all the edge cases-especially with reporting the attendance statistics, as well as amending an attendance record to reflect someone who left early.

### CODE REVIEW

File: app/controllers/attendance_records_controller.rb [hunk 1]
```diff
@@ -19,6 +19,17 @@ def create
     end
   end
 
+  def update
+    @attendance_record = AttendanceRecord.find(params[:id])
+    @attendance_record.sign_out
```
Comment (michaelrkn) [inline app/controllers/attendance_records_controller.rb]: params should look like `{:attendance_record => {:sign_out => true, :student_id => 123}}`. then you can say `@attendance_record.update(attendance_record_params)` and let the model do the work.

File: app/controllers/attendance_records_controller.rb [hunk 2]
```diff
@@ -28,6 +39,6 @@ def destroy
 private
 
   def attendance_record_params
-    params.require(:attendance_record).permit(:student_id)
+    params.require(:attendance_record).permit(:student_id, :left_early, :signed_out_time)
```
Comment (michaelrkn) [inline app/controllers/attendance_records_controller.rb]: :(

File: app/models/attendance_record.rb [hunk 3]
```diff
@@ -5,14 +5,24 @@ class AttendanceRecord < ActiveRecord::Base
 
   before_validation :set_date_and_tardiness
   belongs_to :student
+
```
Comment (michaelrkn) [inline app/models/attendance_record.rb]: add `attr_accessor :signing_out`

Comment (michaelrkn) [inline app/models/attendance_record.rb]: add `before_update` to run `sign_out` if `signing_out == true`

File: app/models/attendance_record.rb [hunk 4]
```diff
@@ -5,14 +5,24 @@ class AttendanceRecord < ActiveRecord::Base
 
   before_validation :set_date_and_tardiness
   belongs_to :student
+  
+  def sign_out
```
Comment (michaelrkn) [inline app/models/attendance_record.rb]: make private

File: app/models/student.rb [hunk 5]
```diff
@@ -44,6 +44,12 @@ def signed_in_today?
     attendance_records.today.exists?
   end
 
+  def signed_out_today?
+    if attendance_records.today.exists?
+      attendance_records.today.first.signed_out_time != nil
+    end
```
Comment (michaelrkn) [inline app/models/student.rb]: use signed_in_today?

File: app/views/attendance_records/index.html.erb [hunk 6]
```diff
@@ -7,6 +7,7 @@
 </div>
 
 <div class="row">
+<% if Time.now.hour < 12 %>
```
Comment (michaelrkn) [inline app/views/attendance_records/index.html.erb]: show one part if signed in, one if not

File: spec/models/attendance_record_spec.rb [hunk 7]
```diff
@@ -52,4 +52,38 @@
       end
     end
   end
+
+  describe '#sign_out' do
```
Comment (michaelrkn) [inline spec/models/attendance_record_spec.rb]: describe "left_early"

**Comment by dustinbrownman** [comment]
What's the purpose of the `AttendanceRecord#signed_out_time`? I ask because it seems weird to keep track of the time they sign out, while not keeping track of when they sign in (though I guess `created_at` technically represents that). Are we using `signed_out_time` to calculate something?

File: app/models/attendance_record.rb [hunk 8]
```diff
@@ -1,18 +1,30 @@
...
 
 private
 
-  def set_tardiness
+  def sign_out
+    class_end_time = Time.zone.parse(ENV['CLASS_END_TIME'] ||= '4:30 PM')
+    current_time = Time.zone.now
+    if current_time >= class_end_time
+      self.left_early = false
+    end
```
Comment (michaelrkn) [inline app/models/attendance_record.rb]: what about `self.left_early = (current_time < class_end_time)`

Comment (michaelrkn) [inline app/models/attendance_record.rb]: or even  `self.left_early = Time.zone.now < Time.zone.parse(ENV['CLASS_END_TIME'] ||= '4:30 PM')`

**Pull request merged**